Code/Source/Lidar/PipelineGraph.h

```diff
@@ -93,10 +105,27 @@ namespace RGL
             rgl_node_t m_parent, m_child;
         };
 
-        static const std::vector<rgl_field_t> DefaultFields;
-
         [[nodiscard]] bool IsFeatureEnabled(PipelineFeatureFlags feature) const;
 
+        //! Get a raycast result of specified field.
+        //! @param results Raycast field result vector.
+        //! @return If successful returns true, otherwise returns false.
+        template<typename FieldType>
+        bool GetResult(AZStd::vector<FieldType>& result, rgl_field_t rglFieldType)
+        {
+            int32_t resultSize = -1;
+            RGL_CHECK(rgl_graph_get_result_size(m_nodes.m_pointsYield, rglFieldType, &resultSize, nullptr));
+
+            if (resultSize <= 0)
+            {
+                return false;
+            }
+
+            result.resize(resultSize);
+            RGL_CHECK(rgl_graph_get_result_data(m_nodes.m_pointsYield, rglFieldType, result.data()));
```

Comment:
What if this call fails? I would make the return value of `GetResult` dependent on `rgl_graph_get_result_data` success.